### DISCUSSION TITLE
0.1.0-rc2 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#371] [Announcements]
This is the second **release candidate** for version **0.1.0**.

This is largely the same as the first release candidate, with the only notable difference being that this one targets Godot 4.0.3.

The list of known issues is carried over unmodified from the first release candidate:

- Extension classes cause exception when used from C# ([#270](https://github.com/godot-jolt/godot-jolt/issues/270))
- Using `DISABLE_MODE_REMOVE` with `PhysicalBone3D` causes a crash ([#333](https://github.com/godot-jolt/godot-jolt/issues/333))
- Changing the `input_ray_pickable` property makes no difference ([#336](https://github.com/godot-jolt/godot-jolt/issues/336))
- Changing the `recovery_as_collision` parameter makes no difference ([#337](https://github.com/godot-jolt/godot-jolt/issues/337))

This release is confirmed to work with Godot 4.0.3 and may or may not be compatible with future versions of Godot.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.1.0-rc2'>0.1.0-rc2</a>.</em>

**Comment by nikitabel229** [comment]
Just tested this extension out, it's AMAZING! Made a quick stress test with small bouncy cubes getting added every frame and falling on a plane, it can easily get 2x more cubes than GodotPhysics before slowing down while also being more stable! Thanks a lot for making this!

**Reply by mihe** [reply]
Happy to hear it! Let me know if you end up using it for something. I'm curious to see what people do with it.

**Reply by nikitabel229** [reply]
I think I'll just try using it in my projects from now on, doesn't seem like it's missing anything that I need, though I'm still a novice so I probably won't make anything worthwhile yet

**Comment by MightyPrinny** [comment]
Ouch, debian stable's glibc is too old for this build, I might have to build from source, but the new stable is supposed to come out in a few months so it might not be that important, just something to consider for linux builds since you get broader compatibility if you compile with an older glibc..

**Reply by mihe** [reply]
Thanks for letting me know. That's a bit of a bummer, considering the lengths I've gone to now in order to link with libgcc and libstdc++ statically, for the express purpose of portability.

Since I build using GitHub Actions I'm sort of bound by the versions of Ubuntu they provide. Technically I could drop back to Ubuntu 20.04 (until they remove it), which might have an old enough glibc for Debian Stable, but from what I can tell that means I could no longer use some of the newer toolchains, like GCC 12, and would have to build them from source, which I'm not likely to do.

There's also the option of switching out glibc with something like musl instead, but I have no idea what sort of knock-on effects that would have. I guess libgcc and libstdc++ would have to be rebuilt using musl as well then, which I'm not likely to do.

I'm also seeing some people use GitHub Actions ability to run Docker containers to somehow provide an environment with an older version of glibc, so that might be something worth exploring.

If you have any other suggestions I'd love to hear them.

**Reply by mihe** [reply]
**tl;dr: The next release should hopefully work with Debian Stable.**

So apparently LLVM does offer a repository of its latest toolchains for Ubuntu 20.04, and since that's the toolchain used during packaging I guess that would make Ubuntu 20.04 an acceptable environment to run in. I could always switch out libstdc++ for libc++ if I ever need something newer from the standard library.

Judging by the version of glibc that's used in Debian Stable, which seems to be 2.31, that should add support for it, since Ubuntu 20.04 also uses glibc 2.31.

If that doesn't help, then there's not much I can do at the moment. The proper solution to this would probably be to make a Docker container that includes a version of LLVM built using musl, and then statically link against musl as well as libc++. That should make the binary entirely self-contained, but that's a rabbit hole I'm not willing to go down in the near-term.
